### PR TITLE
add input, edit some fields, delete record, fetch

### PR DESCRIPTION
**Pull request by ikirienko-exceedteam**

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -95,53 +123,114 @@ const onEnterDownAddCost = (event) => {
 };
 
 const render = () => {
+  console.log('render', records, flags.editInfo);
```
Comment (usatovaDasha) [inline script.js:126]: remove console.log

File: script.js [hunk 2]
```diff
@@ -155,28 +244,252 @@ const onEnterDownInput = (event, index) => {
   }
 };
 
+const onClickEdit = async (event, index, editAll = false) => {
+  console.log('onClickEdit');
```
Comment (usatovaDasha) [inline script.js:248]: remove

File: script.js [hunk 3]
```diff
@@ -155,28 +244,252 @@ const onEnterDownInput = (event, index) => {
...
+  // if nothing is editing yet -> just set editing info
+  if (_.isEmpty(flags.editInfo)) {
+    flags.checkBoxEditSaveDelCancelProcessed = false;
+    setEditingInfo(event.target, index, editAll);
+    render();
+    return;
```
Comment (usatovaDasha) [inline script.js:254]: no need return

File: script.js [hunk 4]
```diff
@@ -155,28 +244,252 @@ const onEnterDownInput = (event, index) => {
...
+  flags.editInfo.index = index;
+
+  if (editAll) {
+    flags.editInfo.editAll = true;
+  } else {
+    flags.editInfo.field = elem.className;
+  }
+};
+
+// const onClickEditButton = async (index) => {
```
Comment (usatovaDasha) [inline script.js:276]: remove comments blocks in file

File: script.js [hunk 5]
```diff
@@ -155,28 +244,252 @@ const onEnterDownInput = (event, index) => {
...
+//   }
+
+//   flags.checkBoxEditSaveDelCancelProcessed = true;
+//   await setValueForEditingTask(flags.editingTaskId);
+//   flags.editingTaskId = index;
+//   render();
+// };
+
 const onClickSaveButton = async (index) => {
+  console.log('onClickSaveButton');
```
Comment (usatovaDasha) [inline script.js:291]: remove

File: script.js [hunk 6]
```diff
@@ -155,28 +244,252 @@ const onEnterDownInput = (event, index) => {
...
+  flags.checkBoxEditSaveDelCancelProcessed = true;
+  await updateValuesForEditingFields();
+  flags.editInfo = {};
+  render();
+};
+
+const onClickCancelButton = (index) => {
+  console.log('onClickCancelButton');
```
Comment (usatovaDasha) [inline script.js:299]: higher

File: script.js [hunk 7]
```diff
@@ -155,28 +244,252 @@ const onEnterDownInput = (event, index) => {
...
+  flags.checkBoxEditSaveDelCancelProcessed = true;
+  flags.editInfo = {};
+  render();
+};
+
+const updateValuesForEditingFields = async () => {
+  console.log('updateValuesForEditingFields');
```
Comment (usatovaDasha) [inline script.js:306]: higher

File: script.js [hunk 8]
```diff
@@ -155,28 +244,252 @@ const onEnterDownInput = (event, index) => {
...
+
+  const {index, editAll, field} = flags.editInfo;
```
Comment (usatovaDasha) [inline script.js:308]: missing spaces

File: script.js [hunk 9]
```diff
@@ -155,28 +244,252 @@ const onEnterDownInput = (event, index) => {
...
+  let newValues = {};
+
+  if (editAll) {
+    // don't include index field
+    fields.slice(1).forEach((field) => {
+      const newValue = getNewValueForField(field, index);
+      console.log('newValue', newValue);
```
Comment (usatovaDasha) [inline script.js:315]: remove

File: script.js [hunk 10]
```diff
@@ -155,28 +244,252 @@ const onEnterDownInput = (event, index) => {
...
+  // }
+  // const jsonBody = JSON.stringify(curRecord);
+
+  newValues._id = records[index]._id;
+  const jsonBody = JSON.stringify(newValues);
+
+  return sendPatchRequest(jsonBody);
+};
+
+const isEveryKeyEqual = (obj1, obj2) => {
```
Comment (usatovaDasha) [inline script.js:344]: you can you lodash functions

File: script.js [hunk 11]
```diff
@@ -155,28 +244,252 @@ const onEnterDownInput = (event, index) => {
...
+    }
+  }
+  return true;
+};
+
+const getNewValueForField = (field, index) => {
+  const curRecValues = records[index];
+  let newValue = null;
+  const editElem = document.querySelector(`#cost-accounting .${getRecordIndex(index)} .${field}`);
+  if (fieldTypes[field] === 'date') {
```
Comment (usatovaDasha) [inline script.js:357]: change on switch case

File: script.js [hunk 12]
```diff
@@ -155,28 +244,252 @@ const onEnterDownInput = (event, index) => {
...
+    
+  } else if (fieldTypes[field] === 'number') {
+    newValue = +editElem.value;
+    // if cost is empty or less than 0 -> use the previous value
+    if (isEmpty(editElem.value) || newValue < 0) {
+      newValue = curRecValues.cost;
+    }
+  } else if (field === 'shop') {
+    if (isEmpty(editElem.value)) {
+      newValue = curRecValues.shop;
```
Comment (usatovaDasha) [inline script.js:381]: change on ternary

File: style.css [hunk 13]
```diff
@@ -1,12 +1,26 @@
...
+  background-color: #303297;
   /* overflow: hidden; */
 }
 
-#cost-accounting input {
-  font-family: 'Noto Serif', serif;
+#cost-accounting input,
+#cost-accounting textarea,
+#cost-accounting p {
+font-family: 'Open Sans', sans-serif;
```
Comment (usatovaDasha) [inline style.css:10]: add tabs

File: style.css [hunk 14]
```diff
@@ -17,21 +31,101 @@ body {
...
+#cost-accounting:after {
+  content: '';
+  position: absolute;
+  z-index: -1;
+  top: -50%;
+  left: 50%;
+  width: 40%;
+  height: 150%;
+  transform: rotate(
+45deg
```
Comment (usatovaDasha) [inline style.css:99]: in one line

File: style.css [hunk 15]
```diff
@@ -17,21 +31,101 @@ body {
...
+}
+
+#cost-accounting:after {
+  content: '';
+  position: absolute;
+  z-index: -1;
+  top: -50%;
+  left: 50%;
+  width: 40%;
+  height: 150%;
```
Comment (usatovaDasha) [inline style.css:97]: 150%??

File: style.css [hunk 16]
```diff
@@ -77,6 +179,112 @@ body {
...
+#cost-accounting .del-all-button:hover {
+  color: #e54b4d;
+  border-color: #fff;
+  background-color: #fff;
+}
+
+#del {
+  align-self: center;
+}
+
```
Comment (usatovaDasha) [inline style.css:288]: remove empty line

File: style.css (hunk 15 above)
Comment (ikirienko-exceedteam) [reply style.css:87]: This is a decorative tilted block.

**Pull request merged**